packages/flutter/lib/src/rendering/object.dart

```diff
@@ -118,25 +118,33 @@ class PaintingContext extends ClipContext {
       return true;
     }());
     OffsetLayer? childLayer = child._layerHandle.layer as OffsetLayer?;
+    final Offset? oldOffset = childLayer?.offset;
```

Comment:
Since it is only used in an assert, grab this only if asserts are enabled?

Reply:
move this inside the else branch?

Reply:
done